src/components/tests/negotiations-panel.test.tsx

```diff
@@ -0,0 +1,141 @@
+            defunct: false,
+            rejected: false,
+            forfeited: false 
+          }
+        },
+        "2": {
+          hasTerminal: false,
+          base: { 
+            sequence: [2],
+            defunct: false,
+            rejected: false,
+            forfeited: true 
+          },
+          terminal: {
+            sequence: [1, 2],
+            defunct: true,
+            rejected: false,
+            forfeited: false 
+          }
+        },
+      },
+      resolved: ResolveState.RESOLVED
+    }
+  }
+});
+
+it('renders negotiations correctly', () => {
+  const root = shallow(<NegotiationsPanel 
+    conflicts={negotiationStatuses}
+    spotlight={undefined}
+    mapFloorLayerSorted={undefined}
+    negotiationStatusManager={undefined}
+    negotiationTrajStore={undefined} />);
+
+  const treeView = root.find(TreeView);
+  expect(treeView).toBeDefined();
+  
+  const treeItem = root.find(TreeItem);
+  expect(treeItem).toBeDefined();
+  expect(treeItem).toHaveLength(4);
+  
+  {
+    const label = treeItem.at(0).prop("label");
+    expect(label).toBeDefined();
+    expect(label).toContain("Conflict");
+    
+    const classes = treeItem.at(0).prop("classes");
+    expect(classes).toBeDefined();
+    expect(classes?.label?.includes("finished"));
+  }
+  
+  {
+    const label = treeItem.at(1).prop("label");
+    expect(label).toBeDefined();
+    expect(label).toContain("[FINISHED]");
+
+    const classes = treeItem.at(1).prop("classes");
+    expect(classes).toBeDefined();
+    expect(classes?.label?.includes("finished"));
+  }
+
+  {
+    const label = treeItem.at(2).prop("label");
+    expect(label).toBeDefined();
+    expect(label).toContain("[REJECTED]");
+
+    const classes = treeItem.at(2).prop("classes");
+    expect(classes).toBeDefined();
+    expect(classes?.label?.includes("rejected"));
+  }
+  
+  {
+    const label = treeItem.at(3).prop("label");
+    expect(label).toBeDefined();
+    expect(label).toContain("[FORFEITED]");
+
+    const classes = treeItem.at(3).prop("classes");
+    expect(classes).toBeDefined();
+    expect(classes?.label?.includes("forfeited"));
+  }
+
+  treeItem.at(1).simulate('click');
+
+  root.unmount();
+});
+
+it('matches snapshot', () => {
+  const root = mount(<NegotiationsPanel 
+    conflicts={negotiationStatuses}
+    spotlight={undefined}
+    mapFloorLayerSorted={undefined}
+    negotiationStatusManager={undefined}
+    negotiationTrajStore={undefined} />);
+
+  expect(toJson(root)).toMatchSnapshot();
+
+  root.unmount();
+});
+
+it ('tests negotiation status manager', () => {
+  let negotiationStatusManager : NegotiationStatusManager;
+  negotiationStatusManager = new NegotiationStatusManager("");
+
+  let conflicts = negotiationStatusManager.allConflicts();
+  expect(conflicts).toBeDefined();
+
+  negotiationStatusManager.removeOldConflicts();
+  let conflicts2 = negotiationStatusManager.allConflicts();
+  expect(conflicts2).toBeDefined();
+});
```

Comment:
Can you move this to another file? 
Since you testing another file, and another class that is not a React component, you can create a `tests/ `directory in `src/tests/ ` and place it there with the following name:` negotiation-status-manager.test.ts`.

Reply:
so i wrote this hoping to see the effects of the codecov bot. it only tests a couple of lines to little benefit; the rest needs to be tested e2e.

while there is that urge to keep everything modular, moving it to another file would bloat the file count which we already have a large number of. i don't see a high count of functions with `negotiation-status-manager` that could warrant such a move so I'd say keep it here.

Reply:
> so i wrote this hoping to see the effects of the codecov bot. it only tests a couple of lines to little benefit; the rest needs to be tested e2e.

Oh ok!



> while there is that urge to keep everything modular, moving it to another file would bloat the file count which we already have a large number of. i don't see a high count of functions with `negotiation-status-manager` that could warrant such a move so I'd say keep it here.

Yeah, I guess it's not worth it to move it. You can leave it as it is.



> file count which we already have a large number of.

Haha Yeah! I feel your pain, I wouldn't say I liked it in the beginning too, but the idea of having a file per component/class and a test class, it helps in the long run.

Reply:
> the idea of having a file per component/class and a test class, it helps in the long run.

I generally disagree with that given all the tools we have for searching code nowadays, but let's move on.